src/discussions/common/HoverCard.test.jsx

```diff
@@ -117,69 +122,58 @@ describe('HoverCard', () => {
     store = initializeStore();
     Factory.resetAll();
     axiosMock = new MockAdapter(getAuthenticatedHttpClient());
-    axiosMock.onGet(threadsApiUrl)
-      .reply(200, Factory.build('threadsResult'));
-    axiosMock.onPatch(new RegExp(`${commentsApiUrl}*`)).reply(({
-      url,
-      data,
-    }) => {
+    axiosMock.onGet(threadsApiUrl).reply(200, Factory.build('threadsResult'));
+    axiosMock.onPatch(new RegExp(`${commentsApiUrl}*`)).reply(({ url, data }) => {
       const commentId = url.match(/comments\/(?<id>[a-z1-9-]+)\//).groups.id;
-      const {
-        rawBody,
-      } = camelCaseObject(JSON.parse(data));
+      const { rawBody } = camelCaseObject(JSON.parse(data));
       return [200, Factory.build('comment', {
         id: commentId,
         rendered_body: rawBody,
         raw_body: rawBody,
       })];
     });
-    axiosMock.onPost(commentsApiUrl)
-      .reply(({ data }) => {
-        const {
-          rawBody,
-          threadId,
-        } = camelCaseObject(JSON.parse(data));
-        return [200, Factory.build(
-          'comment',
-          {
-            rendered_body: rawBody,
-            raw_body: rawBody,
-            thread_id: threadId,
-          },
-        )];
-      });
-
-    executeThunk(fetchThreads(courseId), store.dispatch, store.getState);
-    mockAxiosReturnPagedComments();
-    mockAxiosReturnPagedCommentsResponses();
+    axiosMock.onPost(commentsApiUrl).reply(({ data }) => {
+      const { rawBody, threadId } = camelCaseObject(JSON.parse(data));
+      return [200, Factory.build('comment', {
+        rendered_body: rawBody,
+        raw_body: rawBody,
+        thread_id: threadId,
+      })];
+    });
+
+    await executeThunk(fetchThreads(courseId), store.dispatch, store.getState);
+    await mockAxiosReturnPagedComments();
+    await mockAxiosReturnPagedCommentsResponses();
   });
 
   test('it should have hover card on post', async () => {
-    renderComponent(discussionPostId);
+    await waitFor(() => renderComponent(discussionPostId));
     const post = screen.getByTestId('post-thread-1');
     expect(within(post).getByTestId('hover-card-thread-1')).toBeInTheDocument();
   });
 
   test('it should have hover card on comment', async () => {
-    renderComponent(discussionPostId);
+    await waitFor(() => renderComponent(discussionPostId));
     const comment = await waitFor(() => screen.findByTestId('comment-comment-1'));
     expect(within(comment).getByTestId('hover-card-comment-1')).toBeInTheDocument();
   });
 
   test('it should show add response, like, follow and actions menu for hovered post', async () => {
-    renderComponent(discussionPostId);
+    await waitFor(() => renderComponent(discussionPostId));
     const post = screen.getByTestId('post-thread-1');
     const view = within(post).getByTestId('hover-card-thread-1');
```

Comment:
instead of view it should be menu, just need to replace the view title with menu